lib/models/addon-discovery.js

```diff
@@ -11,7 +11,10 @@ const path = require('path');
 const resolve = require('resolve');
 const findup = require('find-up');
 const heimdall = require('heimdalljs');
-
+const PACKAGE_CACHE = Object.create(null);
```

Comment:
It isn't about this file itself being required though, its this caching currently assumes that during the life of the process none of the addon discovery can change for the same input paths.  This isn't true when (for example) using something like our own test runner where we would generate / regenerate files in node_modules with different contents (e.g. via `input.write` from `broccoli-test-helper`).